src/parser.rs

```diff
@@ -373,7 +377,11 @@ impl Parser {
             Token::LCurly => self.record_or_closure(),
             Token::LParen => {
                 self.tokens.advance();
-                let output = self.expression();
+                let output = if self.tokens.peek().0 == Token::RParen {
```

Comment:
You can use `peek_token()` to get just the token, it's more readable than random `.0` scattered around the code.